commit 72f813e68ee6dbedacc87b98286c4803af30ac76
Author: Tomas Hoger <thoger@fedoraproject.org>
Date:   Wed Apr 10 22:58:35 2013 +0200

    Fix spelling error: endding -> ending
    
    Reported in:
    http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=621698
    
    Original fix by:
    David Paleino <dapal@debian.org>
    
    This updated fix makes macchanger to accept both --ending and --endding
    for backwards compatibility.  Only --ending is documented in help outage
    and man page.

diff --git a/doc/macchanger.texi b/doc/macchanger.texi
index f9cf641..3ec5734 100644
--- a/doc/macchanger.texi
+++ b/doc/macchanger.texi
@@ -115,8 +115,8 @@ Print the current MAC address and exit.
 
 @item -e
 @cindex @code{-e}
-@itemx --endding
-@cindex @code{--endding}
+@itemx --ending
+@cindex @code{--ending}
 Change the MAC address setting the same hardware vendor as the current.
 
 @item -a
diff --git a/macchanger.1 b/macchanger.1
index 549ac0d..3a41a3f 100644
--- a/macchanger.1
+++ b/macchanger.1
@@ -45,7 +45,7 @@ Show summary of options.
 .B \-V, \-\-version
 Show version of program.
 .TP
-.B \-e, \-\-endding
+.B \-e, \-\-ending
 Don't change the vendor bytes.
 .TP
 .B \-a, \-\-another
diff --git a/src/main.c b/src/main.c
index 48e5c0f..d6c757d 100644
--- a/src/main.c
+++ b/src/main.c
@@ -51,7 +51,7 @@ print_help (void)
 		"  -h,  --help                   Print this help\n"
 		"  -V,  --version                Print version and exit\n"
 		"  -s,  --show                   Print the MAC address and exit\n"
-		"  -e,  --endding                Don't change the vendor bytes\n"
+		"  -e,  --ending                 Don't change the vendor bytes\n"
 		"  -a,  --another                Set random vendor MAC of the same kind\n"
 		"  -A                            Set random vendor MAC of any kind\n"
 		"  -p,  --permanent              Reset to original, permanent hardware MAC\n"
@@ -112,7 +112,7 @@ int
 main (int argc, char *argv[])
 {
 	char random       = 0;
-	char endding      = 0;
+	char ending       = 0;
 	char another_any  = 0;
 	char another_same = 0;
 	char permanent    = 0;
@@ -127,7 +127,8 @@ main (int argc, char *argv[])
 		{"help",        no_argument,       NULL, 'h'},
 		{"version",     no_argument,       NULL, 'V'},
 		{"random",      no_argument,       NULL, 'r'},
-		{"endding",     no_argument,       NULL, 'e'},
+		{"ending",      no_argument,       NULL, 'e'},
+		{"endding",     no_argument,       NULL, 'e'}, /* kept for backwards compatibility */
 		{"another",     no_argument,       NULL, 'a'},
 		{"permanent",   no_argument,       NULL, 'p'},
 		{"show",        no_argument,       NULL, 's'},
@@ -166,7 +167,7 @@ main (int argc, char *argv[])
 			random = 1;
 			break;
 		case 'e':
-			endding = 1;
+			ending = 1;
 			break;
 		case 'b':
 			set_bia = 1;
@@ -238,7 +239,7 @@ main (int argc, char *argv[])
 		}
 	} else if (random) {
 		mc_mac_random (mac_faked, 6, set_bia);
-	} else if (endding) {
+	} else if (ending) {
 		mc_mac_random (mac_faked, 3, set_bia);
 	} else if (another_same) {
 		val = mc_maclist_is_wireless (mac);
